src/core/time.ts

```diff
@@ -118,6 +118,19 @@ export class TimeValue {
    * @param other The time value to add to this one.
    */
   add(other: TimeValue): TimeValue {
+    if (
+      (this.isNever() && other.isForever()) ||
```

Comment:
We don't have to treat this case specially, I think.
We have
```
static never(): TimeValue {
    return new TimeValue(Number.MIN_SAFE_INTEGER, 0);
  }

  static forever(): TimeValue {
    return new TimeValue(Number.MAX_SAFE_INTEGER, 0);
  }
```
And `Number.MAX_SAFE_INTEGER` + `Number.MIN_SAFE_INTEGER` = `0`.

Reply:
Never mind this comment. What you did is correct. I've added some unit tests to verify.